qa/L0_backend_python/python_test.py

```diff
@@ -265,7 +265,7 @@ def test_unicode(self):
         with httpclient.InferenceServerClient("localhost:8000") as client:
             utf8 = '😀'
             input_data = np.array([bytes(utf8, encoding='utf-8')],
-                                  dtype=np.bytes_)
+                                  dtype=np.object_)
```

Comment:
What is the usage required? Using np.object in all cases doesn't work, at least not based on findings here: https://jirasw.nvidia.com/browse/DLIS-1955?focusedCommentId=2855169&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-2855169

From that it seems like using np.object will a binary str will cause it to be serialized as the string representation of the binary, which is not what is wanted. The fix is to use np.bytes but that will string ending zeros so shouldn't it really be np.void. We need some testing for all these cases and to make sure that the wire representation of these is the actual binary data (or correct string data) and not some other representation.

Reply:
I understand the issue in DLIS-1955. The problem there is also addressed as a part of this PR. I originally tried to use the `np.void` as we discussed. But the problem with `np.void` was that it raised the error below after serialization:

```
Invalid type promotion with void datatypes of different lengths. Use the `np.bytes_` datatype instead to pad the shorter value with trailing zero bytes.
```

The problem in DLIS-1955 is fixed in the comment I added below. I will add more tests to L0_string_io to make sure all different variations work.